### PR TITLE
erasure: Remove prefix based listing support on ListMultipartUploads

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Previously we have removed this support under FS on #4996,
deprecate this on erasure coded backend as well to simplify.

## Motivation and Context
This is needed as part of https://github.com/minio/minio/projects/14 

## How Has This Been Tested?
Manually and using `mc ls -I` 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by krishnasrinivas — CHANGES_REQUESTED** [review]
@harshavardhana ideally cleanup code will also needs to be part of the PR, if not, the cleanup code should also go in before the release. Else multipart files will never get deleted.

**Comment by harshavardhana** [comment]
> @harshavardhana ideally cleanup code will also needs to be part of the PR, if not, the cleanup code should also go in before the release. Else multipart files will never get deleted.

I thought we want to do cleanup after this PR, since they are both separate functionalities. Since abort is still supported in the API.

**Review by krishnasrinivas — APPROVED** [review]

File: cmd/xl-v1-multipart.go [hunk 1]
```diff
@@ -272,189 +272,54 @@ func commitXLMetadata(disks []StorageAPI, srcBucket, srcPrefix, dstBucket, dstPr
...
-		} else {
-			uploadID = upload.UploadID
-			objectName = upload.Object
+
+		result.NextKeyMarker = object
+		// Loop through all the received uploads fill in the multiparts result.
+		for _, upload := range uploads {
+			uploadID := upload.UploadID
 			result.Uploads = append(result.Uploads, upload)
+			result.NextUploadIDMarker = uploadID
```
Comment (poornas) [inline cmd/xl-v1-multipart.go:310]: you are not restricting total uploads across disks to maxUploads. Should decrement len(uploads) from maxUploads after each iteration.

**Review by poornas — CHANGES_REQUESTED** [review]

File: cmd/xl-v1-multipart.go (hunk 1 above)
Comment (poornas) [reply cmd/xl-v1-multipart.go:310]: not an issue -because of the break statement on line 319

**Review by poornas — APPROVED** [review]

**Pull request merged**